yt_dlp/extractor/adultswim.py

```diff
@@ -196,3 +205,154 @@ def _real_extract(self, url):
             return self.playlist_result(
                 entries, show_path, show_data.get('title'),
                 strip_or_none(show_data.get('metaDescription')))
+
+
+class AdultSwimStreamIE(InfoExtractor):
+    IE_NAME = 'adultswim:stream'
+    _VALID_URL = r'https?://(?:www\.)?adultswim\.com/streams/(?P<id>[^/?#]+)'
+
+    _TESTS = [{
+        'url': 'https://www.adultswim.com/streams/rick-and-morty',
+        'info_dict': {
+            'id': 'rick-and-morty',
+            'title': 'Rick and Morty',
+            'description': 'An infinite loop of Rick and Morty. You\'re welcome. (Marathon available in select regions)',
+        },
+        'params': {
+            # m3u8 download
+            'skip_download': True,
+        },
+        'playlist_mincount': 40,
+    }]
+
+    def _live_hls_fragments(self, episode_start_time, episode_duration, video_id, hls_url, content):
+        FRAGMENT_DURATION = 10.010
+
+        sleep_until = episode_start_time + min(60, episode_duration)
+        if time.time() - sleep_until < 0:
+            format_dur = lambda dur: '%02d:%02d:%02d' % timetuple_from_msec(dur * 1000)[:-1]
+            last_msg = ''
+
+            def progress(msg):
+                nonlocal last_msg
+                self.to_screen(msg + ' ' * (len(last_msg) - len(msg)) + '\r', skip_eol=True)
+                last_msg = msg
+
+            while time.time() <= sleep_until:
+                progress('Waiting for next episode to air (%s) - Press Ctrl+C to cancel' % format_dur(sleep_until - time.time()))
+                time.sleep(1)
+            progress('')
```

Comment:
_wait_for_video() seems to return without waiting when ie_result type is multi_video and not video.
Will I need to call _wait_for_video() from my segment generator like I did here, or should I modify _wait_for_video() calls so that it gets called for each video in entries?

Reply:
> _wait_for_video() seems to return without waiting when ie_result type is multi_video and not video.

That should not happen. You can fix it if you can. Otherwise I'll have a look later

Reply:
After looking over it I feel like the current implementation of _wait_for_video is not designed with playlists in mind. It will raise ReExtractInfo exception for reruning the extractor, and by that force many unnecessary website downloads to gain information that was already available from the start.

Perhaps to fix that _wait_for_video has to make reextraction optional (true by default) and otherwise run a method supplied in info_dict that will return the formats/url.

Reply:
Maybe we could have a new overwriteable method in the infoextractor `_on_reextract(old_ie_result)` that calls `extract` by default and call it on `ReExtractInfo`. Then the extractor can optionally adjust the extraction process to take less time/make less request/reuse old results. After all we dont want the playlist init logs to be shown multiple times to the user or break something else.

Or we could also just handle the `ReExtractInfo` exception per video in a playlist instead of making it reextract the whole playlist... But that would also require some kind of callable that was supplied by the extractor

What do you think? I can try and implement any of these..

Reply:
Well we could also do something sketchy like reextracting the whole playlist and then filtering out the video with the right id

Reply:
pukkandan, pls look at my recent commit, i have tried to fix wait_for_video for playlists. I have also rewritten the extractor.